### PR TITLE
Perl atlas modules

### PR DESCRIPTION
**Pull request by pinin4fjords**
This is a recipe for the Perl modules at https://github.com/ebi-gene-expression-group/perl-atlas-modules. This was originally written by @suhaibMo, and has been available on our group Anaconda channel at https://anaconda.org/ebi-gene-expression-group/perl-atlas-modules. We'd like to host on Bioconda from now to access all the goodness of CI, Biocontainers etc. 

The build script contains a number of Perl module installs for packages not available in Conda, which will then be packaged along with the code from the source repo. I've had a quick go with the Perl Skeleton for these dependencies with a view to adding them, but it looks like there's some fairly complex version dependency issues to resolve and I couldn't get them to build. For now I've minimised the extra code by making sure all upstream dependencies for the non-Conda modules are present at build time so CPAN doesn't need to install them and they don't get added to the package. 

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/perl-atlas-modules/build.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+PERLLIB="${PREFIX}/lib/perl5/${perl_version}/perl_lib"
+mkdir -p $PERLLIB 
+
+## install modules from CPAN directly as they are no conda packages for these modules
+
+# A little hack to try to nudge Bioconda MacOS CI- which seems to have HOME set
+# to 'UNKNOWN' which disrupts cpanm
+
+if [ "$HOME" = 'UNKNOWN' ] && [ -d "Users/distiller" ]; then
```
Comment (dpryan79) [inline recipes/perl-atlas-modules/build.sh:23]: Try changing this to:

```
if [[ ${HOST} =~ .*darwin.* ]]; then
```

Comment (pinin4fjords) [reply recipes/perl-atlas-modules/build.sh]: Thanks @dpryan79 - fingers crossed!

**Comment by pinin4fjords** [comment]
@BiocondaBot please add label

**Review by dpryan79 — APPROVED** [review]

**Comment by dpryan79** [comment]
@bioconda-bot please merge

**Pull request merged**